lib/cloud_controller/runners/puma_runner.rb

```diff
@@ -36,6 +36,10 @@ def initialize(config, app, logger, periodic_updater, request_logs)
         conf.before_fork do
           Sequel::Model.db.disconnect
         end
+        conf.on_worker_boot do
+          ENV['IS_MAIN'] = 'false'
+          prometheus_updater.update_gauge_metric(:cc_db_connection_pool_timeouts_total, 0, labels: { process_type: 'worker' })
```

Comment:
**question/thought** If I'm understanding correctly, this will aggregate all worker metrics into one metric with the single `worker` label. Given the pools are per process, do we care about db metrics per worker?

Reply:
The prometheus registry stores metrics per process. The metrics will have the `pid` label by default, and won't be aggregated. So in this case we are able to see how many connection pool timeouts were thrown by each process. I only added the `process_type` labels for convenience, so that it is easier in the monitoring system to see which process is a puma worker process and which is a puma main process. Otherwise you'd only see the pid and have to check other metrics like worker_backlog for the pid's of the worker processes.